src/Functions/bitTest.cpp

```diff
@@ -21,12 +22,21 @@ struct BitTestImpl
     static const constexpr bool allow_string_integer = false;
 
     template <typename Result = ResultType>
-    NO_SANITIZE_UNDEFINED static Result apply(A a [[maybe_unused]], B b [[maybe_unused]])
+    static Result apply(A a [[maybe_unused]], B b [[maybe_unused]])
     {
         if constexpr (is_big_int_v<A> || is_big_int_v<B>)
             throw Exception(ErrorCodes::NOT_IMPLEMENTED, "bitTest is not implemented for big integers as second argument");
         else
-            return (typename NumberTraits::ToInteger<A>::Type(a) >> typename NumberTraits::ToInteger<B>::Type(b)) & 1;
+        {
+            typename NumberTraits::ToInteger<A>::Type a_int(a);
+            typename NumberTraits::ToInteger<B>::Type b_int(b);
+            const auto max_position = decltype(b)((8 * sizeof(a)) - 1);
+            if (b > max_position || b < 0)
```

Comment:
Nit: Suggest to do the check (l. 33-36) with `a_int` / `b_int` (instead of `a`/`b`) or keep it as is and move it up before l. 31.